sdk/openai/openai/samples-dev/bringYourOwnData.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * Demonstrates how to use Azure's Bring Your Own Data with Azure OpenAI Chat Completions.
+ *
+ * @summary chat completions with your own data.
+ * @azsdk-weight 100
+ */
+
+import { OpenAIClient, AzureKeyCredential } from "@azure/openai";
+
+// Load the .env file if it exists
+import * as dotenv from "dotenv";
+dotenv.config();
+
+// You will need to set these environment variables or edit the following values
+// The endpoint you will use to access your Azure OpenAI instance
+const endpoint = process.env["ENDPOINT"] || "<endpoint>";
+// Your Azure OpenAI API key
+const azureApiKey = process.env["AZURE_API_KEY"] || "<api key>";
+// Your Azure Cognitive Search endpoint, key, and index name
+const azureSearchEndpoint = process.env["AZURE_SEARCH_ENDPOINT"] || "<search endpoint>";
+const azureSearchKey = process.env["AZURE_SEARCH_KEY"] || "<search key>";
+const azureSearchIndex = process.env["AZURE_SEARCH_INDEX"] || "<search index>";
+
+const messages = [
+  {
+    role: "user",
+    content: "What's the most common feedback we received from our customers about the product?",
+  },
+];
+
+export async function main() {
+  console.log("== Bring Your Own Data Sample ==");
+
+  const client = new OpenAIClient(endpoint, new AzureKeyCredential(azureApiKey));
+  const deploymentId = "gpt-35-turbo";
+  const events = client.listChatCompletions(deploymentId, messages, {
+    maxTokens: 128,
+    azureExtensionOptions: {
+      extensions: [
```

Comment:
Maybe a comment as to what the azure extensions are?  Unless that's covered in the README.

Reply:
Done in https://github.com/Azure/azure-sdk-for-js/pull/26907/commits/c27052ea439aa224df870352f33555d0d77a53d2.